custom/opm/beneficiary.py

```diff
@@ -913,3 +914,175 @@ def __init__(self, case, report, child_index=1, **kwargs):
+                                                       "तीस आयरन की गोलियां नही लिया", self.preg_received_ifa)
+        self.four = self.condition_image(C_ATTENDANCE_Y, C_ATTENDANCE_N, "पोषण दिवस में उपस्थित",
+                                         "पोषण दिवस में उपस्थित नही", self.child_attended_vhnd)
+        self.five = self.condition_image(C_REGISTER_Y, C_REGISTER_N, "जन्म पंजीकृत",
+                                         "जन्म पंजीकृत नहीं", self.child_age == 6)
+        self.six = self.condition_image(C_WEIGHT_Y, C_WEIGHT_N, "बच्चे का वज़न लिया गया",
+                                        "बच्चे का वज़न लिया गया", self.child_growth_calculated)
+        self.seven = self.condition_image(EXCBREASTFED_Y, EXCBREASTFED_N, "केवल माँ का दूध खिलाया गया",
+                                          "केवल माँ का दूध नहीं खिलाया गया", self.child_breastfed)
+        self.eight = self.condition_image(CHILD_WEIGHT_Y, CHILD_WEIGHT_N, "जन्म के समय वजन लिया गया",
+                                          "जन्म के समय वजन नहीं लिया गया", self.child_age == 3)
+        self.nine = self.condition_image(ORSZNTREAT_Y, ORSZNTREAT_N, "", "",
+                                         self.child_received_ors if self.status == 'mother' else False)
+        self.ten = self.condition_image(MEASLEVACC_Y, MEASLEVACC_N, "खसरे का टिका लगाया",
+                                        "खसरे का टिका नहीं लगाया", self.child_age == 12)
+        self.opened_on = self.case_property('opened_on', EMPTY_FIELD)
+        self.closed_on = self.case_property('closed_on', False)
+        self.close_mother_dup = self.get_value_from_form(CLOSE_FORM, 'form/close_mother_dup')
+        self.close_mother_mo = self.get_value_from_form(CLOSE_FORM, 'form/close_mother_mo')
+        self.leave_program = self.get_value_from_form(CLOSE_FORM, 'form/leave_program')
+        self.close_mother_dead = self.get_value_from_form(CLOSE_FORM, 'form/close_mother_dead')
+        # TODO Add correct values
+        self.weight_this_month = EMPTY_FIELD
+        self.nutritional_status_this_month = EMPTY_FIELD
+        self.nutritional_status_bonus = EMPTY_FIELD
+
+    def get_first_or_empty(self, list):
+        return (list[0] or EMPTY_FIELD) if list else EMPTY_FIELD
+
+    def get_value_from_form(self, form_xmlns, path):
+        return self.get_first_or_empty([
+            form.xpath(path)
+            for form in self.filtered_forms(form_xmlns)
+        ])
+
+    @property
+    def dob_known(self):
+        return any(
+            form.xpath('form/dob_known') == 1
+            for form in self.filtered_forms(PREG_REG_XMLNS)
+        )
+
+    @property
+    def mother_age(self):
+        if self.dob_known:
+            path = 'form/age_known'
+        else:
+            path = 'form/age_unknown'
+        return self.get_value_from_form(PREG_REG_XMLNS, path)
+
+    def preg_weighed_trimestered(self, query_preg_month):
```

Comment:
Can this report not make use of `OPMCaseRow`, seems good amount of code-overlap?